pkg/types/types.go

```diff
@@ -231,7 +239,21 @@ func (s byStepTime) Swap(i, j int) {
 }
 
 func (s byStepTime) Less(i, j int) bool {
```

Comment:
I think we can use sort.Slice instead of special type for sorting. It [was introduced in go 1.8](https://go.dev/doc/go1.8#sort_slice). OTOH maybe it should be a separate patch 🤷

Reply:
yep. could be a separate change. and as a separate function also isn't bad, we can reuse it somewhere else if needed.